### PR TITLE
Add site visibility badge to admin menu

### LINKED ISSUE #50777: Move site visibility badge to admin bar
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

We have received feedback to make the site visibility badge easier to find. We'll try moving it from the home screen title to the admin bar.

image

[Figma](https://www.figma.com/design/0BWMerqRR1f468EwC7g5tg/LYS?node-id=3675-34159&t=gQZVBG8flvIj9Qsx-4)

## Acceptance criteria

* Labels are "Coming soon", "Store coming soon", and "Live"
* The badge is black text on #f6f7f7 (off white) when coming soon
* The badge is #00450C text on #E8F1E9 (dark green on pale green) when live
* The tour should point to the new badge location
* No hover treatment is applied

### PR DESCRIPTION
**Pull request by adrianduffell**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #50777 .

Moves the site visibility badge to the admin bar. 

admin bar badge

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

## Badge

1. Go to WooCommerce > Home
2. See that the site visibility badge has moved from the title (Home) to the admin bar
3. Do some exploratory testing of other admin pages. See that the badge appears on every page
4. View the front end as an admin user. See that the badge appears in the admin bar
5. Click the badge
6. See that the site visibility settings screen is loaded
7. Change site visibility settings to live
8. See that the badge displays "Live" in green
9. Change site visibility settings to coming soon and full site
10.  See that the badge displays "Coming soon" in white
11. Change site visibility settings to coming soon and store pages only
10.  See that the badge displays "Store coming soon" in white

### Tour
1. Install WooCommerce 9.0
2. Update to this branch
3. Perform the database update when prompted by banner
4. Go to WooCommerce > Home
5. See that the tour points to the new badge

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Internal/ComingSoon/ComingSoonAdminBarBadge.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	/**
+	 * Invalidate the WordPress object cache and other known caches.
+	 *
+	 * @internal
+	 */
+	public function output_css() {
+		if ( is_admin_bar_showing() ) {
+			echo '<style>
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Internal/ComingSoon/ComingSoonAdminBarBadge.php:64]: I was on the fence whether to embed this or add it to a new file. Either way, we don't want it to be included when the admin bar isn't showing for customers / anon users.

File: plugins/woocommerce/src/Internal/ComingSoon/ComingSoonAdminBarBadge.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+				$key = 'store-coming-soon';
+			} else {
+				$key = 'coming-soon';
+			}
+		} else {
+			$key = 'live';
+		}
+
+		$args = array(
+			'id'    => 'woocommerce-site-status-badge',
```
Comment (rjchow) [inline plugins/woocommerce/src/Internal/ComingSoon/ComingSoonAdminBarBadge.php]: small nit, should it be consistent and be `site-visibility-badge` everywhere?

**Review by rjchow — COMMENTED** [review]
Small nit, otherwise tested well. Also I wonder if we should also show the tour to existing sites with Wordpress that installed WC? In a follow up issue perhaps

**Review by chihsuan — COMMENTED** [review]
Tested well! I made some additional changes:

1. Removed unused props b52548860ce2bbf4173ad429120a2d2c4070cc4f
2. Fixed tour logic e895b58ad582e6f0d5cd3c2a4a3ee756760337b8 
3. Addressed RJ's feedback. Update site visibility badge ID in tour and admin bar 39b503d38db1760a2a0642f01ac11f35dd871dbc

I'll let others to approve this PR.

File: plugins/woocommerce/src/Internal/ComingSoon/ComingSoonAdminBarBadge.php (hunk 2 above)
Comment (chihsuan) [reply plugins/woocommerce/src/Internal/ComingSoon/ComingSoonAdminBarBadge.php]: Updated in 3c4f9fdeebb1d9187e7961d27e4ed20d17ccdd35

**Review by ilyasfoo — APPROVED** [review]
It seems the new classes are run for anonymous, public users as well. I think a minor improvement would be to have a super early exit when users are not logged in. However, this can be a follow-up.

Tests well! LGTM 🚢

**Comment by adrianduffell** [comment]
Latest changes looking great! 🚀

**Comment by ilyasfoo** [comment]
Added a couple of issues for follow-up:

- https://github.com/woocommerce/woocommerce/issues/50787
- https://github.com/woocommerce/woocommerce/issues/50789

**Comment by martymi** [comment]
Not sure if this is the right place to request changes to this particular update, it's causing a lot of headaches for us and our clients. The LIVE display is incredibly distracting to have constantly displayed on the toolbar. It is also confusing as it will display LIVE even on a staging site, and it is inconsistent with other network sites that don't use Woo.

Requests:
1. A way to disable Woo Coming Soon mode entirely, from the options page
2. A way to disable Woo Coming Soon mode entirely, via a code snippet
3. A way to hide the "LIVE" label in the toolbar if the function is not disabled (perhaps only have a label if 'Coming soon' mode is active.

Thanks for the consideration.

**Pull request merged**